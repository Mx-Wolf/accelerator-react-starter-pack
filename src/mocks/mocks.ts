import {datatype, image, name, random} from 'faker';
import {Guitar} from '../types/guitar';

export const makeFakeComment = () => ({
  id: datatype.number(50).toString(),
  userName: name.title(),
  advantages: datatype.string(10),
  disadvantages: datatype.string(10),
  comment: datatype.string(10),
  rating: datatype.number(5),
  createAt: datatype.string(10),
  guitarId: datatype.number(5),
});

export const makeFakeCommentsList = (value: number) => new Array(value).fill(null).map(() => makeFakeComment());


export const makeFakeGuitar = () => ({
  comments: makeFakeCommentsList(4),
  id: datatype.number(50),
  name: name.title(),
  vendorCode: name.title(),
  type: random.word(),
  description: datatype.string(10),
  previewImg: image.image(),
  stringCount: datatype.number(12),
  rating: datatype.number(5),
  price: datatype.number(100),
});

export const makeFakeGuitarsList = (value: number): Guitar[] => new Array(value).fill(null).map(() => makeFakeGuitar());
