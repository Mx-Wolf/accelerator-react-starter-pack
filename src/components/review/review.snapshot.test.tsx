import renderer from 'react-test-renderer';
import Review from './review';
import {fakeComment} from '../../mocks/mocks';

describe('Component: Review', () => {
  it('should render correctly', () => {
    const component = renderer.create(<Review review={fakeComment}/>);
    expect(component).toMatchSnapshot();
  });
});
