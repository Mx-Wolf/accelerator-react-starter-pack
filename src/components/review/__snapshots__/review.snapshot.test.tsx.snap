// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Review should render correctly 1`] = `
<div
  className="review"
>
  <div
    className="review__wrapper"
  >
    <h4
      className="review__title review__title--author title title--lesser"
    >
      name
    </h4>
    <span
      className="review__date"
    >
      Invalid Date
    </span>
  </div>
  <div
    aria-hidden="true"
    className="rate review__rating-panel"
  >
    <span
      className="visually-hidden"
    >
      Рейтинг:
    </span>
    <svg
      aria-hidden="true"
      height="11"
      width="12"
    >
      <use
        xlinkHref="#icon-full-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="11"
      width="12"
    >
      <use
        xlinkHref="#icon-full-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="11"
      width="12"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="11"
      width="12"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="11"
      width="12"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
    <span
      className="rate__count"
    />
    <span
      className="rate__message"
    />
  </div>
  <h4
    className="review__title title title--lesser"
  >
    Достоинства:
  </h4>
  <p
    className="review__value"
  >
    advantage
  </p>
  <h4
    className="review__title title title--lesser"
  >
    Недостатки:
  </h4>
  <p
    className="review__value"
  >
    disadvantage
  </p>
  <h4
    className="review__title title title--lesser"
  >
    Комментарий:
  </h4>
  <p
    className="review__value"
  >
    comment
  </p>
</div>
`;
